services/lumberjack/src/service.js

```diff
@@ -0,0 +1,293 @@
+    const names = await this._influx.getDatabaseNames();
+    if (!names.includes(this._dbName))
+      await this._influx.createDatabase(this._dbName);
+
+    this._startTasks();
+    logger.debug('Service started');
+  }
+
+  /** Stop the service. */
+  async stop() {
+    logger.debug('Stopping service.');
+
+    await Promise.all([
+      this._pingTask.stop(),
+      this._uploadRateTask.stop(),
+      this._groundTelemetryTask.stop(),
+      this._planeTelemetryTask.stop(),
+    ]);
+
+    logger.debug('Service stopped.');
+  }
+
+  _startTasks() {
+    this._pingTask =
+      createTimeoutTask(this._ping.bind(this),
+        this._pingInterval)
+        .on('error', logger.error)
+        .start();
+    this._uploadRateTask =
+      createTimeoutTask(this._uploadRate.bind(this),
+        this._uploadInterval)
+        .on('error', logger.error)
+        .start();
+    this._groundTelemetryTask =
+      createTimeoutTask(this._groundTelemetry.bind(this),
+        this._telemInterval)
+        .on('error', logger.error)
+        .start();
+    this._planeTelemetryTask =
+      createTimeoutTask(this._planeTelemetry.bind(this),
+        this._telemInterval)
+        .on('error', logger.error)
+        .start();
+  }
+
+  /** Get service ping data and write to the database */
+  async _ping() {
+    // Get ping data
+    let ping;
+    try {
+      ping =
+        (await request.get('http://' + this._pingHost + ':' +
+          this._pingPort + '/api/ping')
+          .proto(stats.PingTimes)
+          .timeout(1000)).body;
+    } catch (err) {
+      logger.error(err);
+      await this._influx.writeMeasurement('ping', [
+        {
+          fields: { apiPing: 0 },
+          tags: { host: 'non-existent-service', port: 0, name: 'service'}
+        }], {
+        database: this._dbName
+      });
+    }
+
+    // Write data for services
+    for (let endpoint of ping.service_pings) {
+      const { host, port, name } = endpoint;
+      await this._influx.writeMeasurement('ping', [
+        {
+          fields: { apiPing: endpoint.ms },
+          tags: { host, port, name }
+        }], {
+        database: this._dbName
+      });
+    }
+
+    // Write data for devices
+    for (let endpoint of ping.device_pings) {
+      const { host, port, name } = endpoint;
+      await this._influx.writeMeasurement('ping', [
+        {
+          fields: { devicePing: endpoint.ms },
+          tags: { host, port, name }
+        }], {
+        database: this._dbName
+      });
+    }
+  }
+
+  /** Get telemetry upload rate data and write to the database */
+  async _uploadRate() {
+    // Get upload rate
+    let rate;
+    try {
+      rate =
+      (await request.get('http://' + this._forwardInteropHost + ':' +
+        this._forwardInteropPort + '/api/upload-rate')
+        .proto(stats.InteropUploadRate)
+        .timeout(1000)).body;
+    } catch (err) {
+      await this._influx.writeMeasurement('upload-rate', [
```

Comment:
I would set rate to a Protobuf object that has the fields set to zero, that way you don't need to have a separate write call. Also the host and port aren't set correctly in this block. Need to log the error here as well.